src/pasta/Vesta.sol

```diff
@@ -359,25 +360,24 @@ library Vesta {
             z3 := add(z3, sub(doubleP, add(z1z1, z2z2)))
             z3 := mulmod(z3, h, P_MOD)
         }
-        */
 
-        assembly {
```

Comment:
I remember, there was some issues with this assembly block - it produced "Stack is too deep" error in my local environment. May be it was some sort of glitch. If it works now (and it works as I can see by successful CI tests passing), we ca remove the TODO comment above and alternative assembly block below (which is commented now). WDYT?

Reply:
I didn't run into the "Stack too deep" issue when I was using the `add` function, but it may be that I wasn't looking deeply enough. Do you remember what was causing the issue when you ran into it? I can try to experiment to see if it's still happening...

Reply:
TBH, I don't remember the obstacles that causes this issue. Once again, I tried running the tests locally using original version of this assembly block and they passed, so looks like that `foundry` optimises compilation by default. May be it was some sort of glitch or I somehow compiled and run the code without optimizations.